src/batInnerShuffle.ml

```diff
@@ -0,0 +1,12 @@
+let array_shuffle ?state a =
+  let random_int state n = match state with
+    | None -> Random.int n
+    | Some s -> Random.State.int s n in
+  for n = Array.length a - 1 downto 1 do
+    let k = random_int state (n + 1) in
+    if k <> n then
```

Comment:
you should put '(' ')' around this `then`, as the indentation suggests. Here the second `unsafe_set` will always run.